Lib/fontmake/font_project.py

```diff
@@ -420,6 +450,33 @@ def _output_path(self, ufo, ext, is_instance=False, interpolatable=False,
             os.makedirs(out_dir)
         return os.path.join(out_dir, '%s.%s' % (self._font_name(ufo), ext))
 
+    def _designspace_locations(self, designspace_path):
+        """Map font filenames to their locations in a designspace."""
+
+        maps = []
+        for location_list in varLib.designspace.load(designspace_path):
+            location_map = {}
+            for loc in location_list:
+                abspath = os.path.normpath(os.path.join(
+                    os.path.dirname(designspace_path), loc['filename']))
+                location_map[abspath] = loc['location']
+            maps.append(location_map)
+        return maps
+
+    def _closest_location(self, location_map, target):
+        """Return path of font whose location is closest to target."""
+
+        dist = lambda a, b: math.sqrt(sum((a[k] - b[k]) ** 2 for k in a.keys()))
```

Comment:
Yea but the target point would also be normalized. I'm thinking that the normalization factors cancel out in the end and so comparisons of normalized locations yield the same result.

And yep in the typical case we'd expect 2 or 3 axes. Though I think any number is possible with MutatorMath designspaces.